crates/services/producer/src/block_producer/gas_price.rs

```diff
@@ -29,7 +29,10 @@ impl From<BlockHeight> for GasPriceParams {
 /// Interface for retrieving the gas price for a block
 pub trait GasPriceProvider {
     /// The gas price for all transactions in the block.
-    fn gas_price(&self, params: GasPriceParams) -> Option<u64>;
+    fn gas_price(
+        &self,
+        params: GasPriceParams,
+    ) -> Result<u64, Box<dyn std::error::Error>>;
```

Comment:
I guess so. Since our "Producer Error" is `anyhow::Error`. 

There are other places where I might invert the error dep for the adapters as well. 

I'm torn on this in Rust in general, because adapters necessarily have their own error type, so you have to use type erasure. And because of that, I figured we could just put the domain errors on the business side rather than expecting the adapter to return the right error type.

For example:

```rs
        let previous_gas_prices = self
            .gas_price_history
            .gas_prices(latest_block_height)
            .map_err(Error::UnableToGetGasPrices)?
            .ok_or(Error::GasPricesNotFoundForBlockHeight(latest_block_height))?;
```

But that's noisy, and we are already trusting the adapter code to uphold the contract of the port, so maybe we just push into the adapters instead of using this `ForiegnError` construct that I used.